cmd/cosign/cli/options/flags.go

```diff
@@ -44,6 +44,8 @@ func NOf(args ...interface{}) int {
 // Deprecated: this will be deleted when the migration to cobra is finished.
 func ApplyRegistryFlags(regOpts *RegistryOptions, fs *flag.FlagSet) {
 	fs.BoolVar(&regOpts.AllowInsecure, "allow-insecure-registry", false, "whether to allow insecure connections to registries. Don't use this for anything but testing")
+	fs.StringVar(&regOpts.TagPrefix, "signature-prefix", "", "custom prefix to use for signature tag")
+	fs.StringVar(&regOpts.TagSuffix, "signature-suffix", "", "custom suffix to use for signature tag")
```

Comment:
Hmm, these don't really feel like registry flags to me.

I think surfacing a way to customize suffix has come up a few times now, and this totally makes sense.  I think we should probably plumb through a similar change for attestations.

Especially if we leave the field in such a generic place, we should have the field name's specificity match the flag.  I could see `--suffix` setting `TagSuffix` or `--signature-suffix` setting `SignatureTagSuffix`, but as it is seems a bit odd.

Reply:
cc @dekkagaijin @n3wscott  for thoughts on where this lives, and consistency with other flags.

Reply:
@mattmoor I agree, but the exiting suffix code was living in registry options so I extended it to include prefix and exposed the existing `TagSuffix` as a flag as well. `TagPrefix` is more important to me because of ECR Lifecycle Policy tag prefix filtering.

I'll push an update to make the `TagPrefix` signature, SBOM, and attestation agnostic.